src/components/inputs/rich-text-input/rich-text-input.js

```diff
@@ -1,25 +1,73 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import omit from 'lodash/omit';
 import requiredIf from 'react-required-if';
-import Types from 'slate-prop-types';
 import { Editor } from 'slate-react';
 import filterDataAttributes from '../../../utils/filter-data-attributes';
 import renderEditor from './editor';
 import plugins from '../../internals/rich-text-plugins';
 import html from '../../internals/rich-text-utils/html';
 import isEmpty from '../../internals/rich-text-utils/is-empty';
 
-class RichTextInput extends React.PureComponent {
+const omittedPropsForRender = ['value'];
+
+class RichTextInput extends React.Component {
+  state = {
+    // we keep track of the serialized (HTML) value
+    serializedValue: this.props.value || '',
+    internalSlateValue: html.deserialize(this.props.value || ''),
+  };
+
+  shouldComponentUpdate(nextProps, nextState) {
+    // ignore updates for changes to `props.value` because this is
+    // actually an `HTML` value.
+
+    // instead we want to update when the slate `value` changes, which
+    // is stored in state.
+    const props = omit(this.props, omittedPropsForRender);
+
+    const havePropsChanged = Object.entries(props).some(
+      ([key, val]) => nextProps[key] !== val
+    );
+
+    if (havePropsChanged) return true;
+
+    if (this.props.value !== this.state.serializedValue) return true;
+
+    if (this.state.internalSlateValue !== nextState.internalSlateValue)
```

Comment:
Does this check make sense? Internal slate value gonna change whenever user types something, which means previous check would already return `true`.

In general looks like this approach is a bit over-complicated. Wouldn't something like this do the same with the same performance, but with the less complexity:

```js
const propsToState = props => ({
    serializedValue: props.value || '',
    internalSlateValue: html.deserialize(props.value || ''),
}); 

class RichTextInput extends React.PureComponent {
  state = propsToState(this.props);

  static getDerivedStateFromProps(props, state) {
    const nextState = propsToState(props);
    if (nextState.serializedValue == state.serializedValue) return null;
    return nextState;
  }  

...

}
```

without the need to implement `componentDidUpate` and `shouldComponentUpdate`? Or am I missing some use-case?

Reply:
Yeah that seems to accomplish the same result in less confusing code 👍

Reply:
[here](https://github.com/commercetools/ui-kit/pull/1098/commits/ba4134e57e1670f71f166f3410151f8c6f9f6cd7)

Reply:
Thanks, I think this is a lot more clear

Reply:
Nice suggestion. At least it wasn't just my head getting super confused by that logic there just not smart enough to give a suggestion 💃